[Feature] Support show grants for new RBAC

HangyuanLiu opened the pull request:
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：
<!--
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

## Problem Summary(Required) ：
<!-- (Please describe the changes you have made. In which scenarios will this bug be triggered and what measures have you taken to fix the bug?) -->

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

amber-create committed:
[Doc] update aws s3 parameter names in broker load docs (#16927)

Signed-off-by: amber-create <yangyanping@starrocks.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

silverbullet233 committed:
[BugFix] Disable RewriteMinMaxAggToMetaScanRule when table has deletion (#16898)

Signed-off-by: silverbullet233 <3675229+silverbullet233@users.noreply.github.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

ZiheLiu committed:
[BugFix] Init scan executor for compute node (#16933)

Signed-off-by: ZiheLiu <ziheliu1024@gmail.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

ABingHuang committed:
add column prune after mv rewrite

Signed-off-by: ABingHuang <codekhuang@163.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

HangyuanLiu committed:
Support show grants for new RBAC

Signed-off-by: HangyuanLiu <460660596@qq.com>

HangyuanLiu committed:
Support show grants for new RBAC

Signed-off-by: HangyuanLiu <460660596@qq.com>

miomiocat committed:
[Feature] Support INT64 based timestamp for JNI connector (#16920)

Signed-off-by: miomiocat <284487410@qq.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

kangkaisen committed:
[BugFix] Fix bitnot largeint cast (#16928)

Signed-off-by: kangkaisen <kangkaisen@apache.org>
Signed-off-by: HangyuanLiu <460660596@qq.com>

ZiheLiu committed:
[BugFix] Set name of MetaScanOperator (#16936)

Signed-off-by: ZiheLiu <ziheliu1024@gmail.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

decster committed:
[BugFix] Fix error in incremental clone introduced by #15935 (#16930)

PR #15935 introduced a new mechanism that an incremental clone can fall back to a full clone if there are too many rowsets need to be copied. But the fallback code return a NotFound error rather than OK, this will cause clone to fail.

```
I0127 00:46:02.271173  7878 snapshot_manager.cpp:112] make primary snapshot tablet:334728 cur_version:65905 missing_version_ranges:65662 timeout:180
I0127 00:46:02.271206  7878 tablet_updates.cpp:2954] get_rowsets_for_snapshot: too many rowsets for incremental clone #rowset:244 #rowset_for_full_clone:1 tablet:334728 #version:1 [65905.1 65905.1@0 65905.1] #pending:0
W0127 00:46:02.271214  7878 agent_server.cpp:308] fail to make_snapshot. tablet_id:334728 msg:Not found: get_rowsets_for_snapshot: too many rowsets for incremental clone #rowset:244 #rowset_for_full_clone:1 tablet:334728 #version:1 [65905.1 65905.1@0 65905.1] #pending:0
```
This PR fixes this bug.

Signed-off-by: Binglin Chang <decstery@gmail.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

hellolilyliuyi committed:
change-toc-of-create-routine-load (#16943)

Signed-off-by: HangyuanLiu <460660596@qq.com>

xiaoyong-z committed:
[Enhancement] Upgrade broker's aliyun-sdk from 2.7.2 to jindosdk-4.6.2 (#16935)

Signed-off-by: HangyuanLiu <460660596@qq.com>

sduzh committed:
[BugFix] Schema change job cannot be cancelled (#16787)

Fix a lake table's schema change job of state FINISHED_REWRITTING cannot
be cancelled even after the table has been dropped.

Signed-off-by: sduzh <zhuming9011@qq.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

imay commented on fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4:
```diff
@@ -1253,13 +1248,13 @@ revokePrivilegeStatement
 grantRoleStatement
-    : GRANT identifierOrString TO user                          #grantRoleToUser
-    | GRANT identifierOrString TO ROLE identifierOrString       #grantRoleToRole
+    : GRANT identifierOrStringList TO user                                                              #grantRoleToUser
```
```suggestion
    : GRANT identifierOrStringList TO user                                                              #grantRoleToUser
    : GRANT identifierOrStringList TO USER user                                                              #grantRoleToUser
```
In the design doc, starrocks support to `grant role to USRE user_name` explicitly.

HangyuanLiu replied:
This PR is only to support the show grants statement. The current RBAC needs to be improved in many details. I will split it into multiple PRs and submit them separately. Each PR should be controlled on one function as much as possible, which is more convenient for review code.

amber-create committed:
[Doc] modify s3 param names (#16960)

Signed-off-by: amber-create <yangyanping@starrocks.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

liuyehcf committed:
[Enhancement] Replace BitSet with RoaringBitMap to lower memory usage (#16499)

Signed-off-by: HangyuanLiu <460660596@qq.com>

GavinMar committed:
[BugFix] Avoid calling `GetReadStatistics` function when accessing juicefs (#16931)


Signed-off-by: GavinMar <yangguansuo@starrocks.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

Commit:
[Doc] Fix Lint errors (#16965)

Signed-off-by: EsoragotoSpirit <wanglichen@starrocks.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

xiaoyong-z committed:
[Enhancement] support broker load from azure abfs (#16760)

In this pr, we support broker load from azure abfs.
We remove the check of path prefix in getFileSystem, if the path prefix is not hdfs/s3a/oss/cos...,
we will try to create a generic file system. The reason why we can do this is because hadoop/s3 SDK
is compatible with all file/object storage system.
Then if in future, if there is a request for supporting new file system,
we only need to add the settings in hdfs-site.xml, core-site.xml of BE/FE, and the jar lib in the specific place.

Signed-off-by: xyz <a997647204@gmail.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

evelynzhaojie committed:
[Doc] trim and create table (#16982)

* [Doc] trim and create table

* Signed-off-by:
evelynzhaojie
<zhaojie@starrocks.com>

* Signed-off-by:
evelynzhaojie
<zhaojie@starrocks.com>

Signed-off-by: HangyuanLiu <460660596@qq.com>

Commit:
[Doc] Add details for BACKUP and MV (#16973)

Signed-off-by: EsoragotoSpirit <wanglichen@starrocks.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

Commit:
[Doc] Fix MV desc and link (#16974)

Signed-off-by: EsoragotoSpirit <wanglichen@starrocks.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

Commit:
[Doc] Fix issues in Config and RG (#16983)

Signed-off-by: EsoragotoSpirit <wanglichen@starrocks.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

nshangyiming committed:
[Feature] Fix submit task privilege error (#16907)

PrivilegeCheckerV2 should also support unqualified table name.

Signed-off-by: HangyuanLiu <460660596@qq.com>

stdpain committed:
[Enhancement] Reduce memory usage in full sorter (#16937)

1. avoid copy and reserve extra chunk when merge sorted chunks
2. release memory in SortRun::reset


Signed-off-by: stdpain <drfeng08@gmail.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

ABingHuang committed:
fix unstable refresh materialized view ut

Signed-off-by: HangyuanLiu <460660596@qq.com>

amber-create committed:
[Doc] update EXPORT (#16992)

Signed-off-by: amber-create <yangyanping@starrocks.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

luohaha committed:
[BugFix] fix be crash when schema change (#16999)

When concurrent run data loading and schema change, BE may crash at:
```
 0x57baa51 starrocks::vectorized::SchemaChangeHandler::_convert_historical_rowsets()
*** SIGSEGV (@0x40) received by PID 13522 (TID 0x7fc2bb3a0700) from PID 64; stack trace: ***
    @          0x6a6b792 google::(anonymous namespace)::FailureSignalHandler()
    @     0x7fc3fd261630 (unknown)
    @          0x57baa51 starrocks::vectorized::SchemaChangeHandler::_convert_historical_rowsets()
    @          0x57bd560 starrocks::vectorized::SchemaChangeHandler::_do_process_alter_tablet_v2_normal()
    @          0x57be888 starrocks::vectorized::SchemaChangeHandler::_do_process_alter_tablet_v2()
    @          0x57bed8a starrocks::vectorized::SchemaChangeHandler::process_alter_tablet_v2()
    @          0x55dde96 starrocks::EngineAlterTabletTask::execute()
    @          0x53faeee starrocks::StorageEngine::execute_task()
    @          0x3a4171e starrocks::run_alter_tablet_task()
    @          0x5b4c405 starrocks::ThreadPool::dispatch_thread()
    @          0x5b4712a starrocks::Thread::supervise_thread()
    @     0x7fc3fd259ea5 start_thread
    @     0x7fc3f293f9fd __clone
    @                0x0 (unknown)
```
Because in _convert_historical_rowsets, assume that sc_params.rowset_readers and sc_params.rowsets_to_change have same valid items. If don't, crash will happen.

So I add checking code when geneate rowsets_to_change, when base_tablet->get_rowset_by_version can't find rowset by version, print all helpful message to check what happen and just return error to prevent BE from crash. Still need find out in which situation, version from capture_consistent_versions by max version can't find rowset by get_rowset_by_version.

Signed-off-by: luohaha <18810541851@163.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

ZiheLiu committed:
[BugFix] Own short_key_encode result for LogicalSplitMorselQueue (#16977)

Signed-off-by: ZiheLiu <ziheliu1024@gmail.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

Vision0220 committed:
Update README.md (#16979)

Signed-off-by: HangyuanLiu <460660596@qq.com>

Youngwb committed:
[UT] Fix unstable ut (#17003)

Signed-off-by: Youngwb <yangwenbo_mailbox@163.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

LiShuMing committed:
Remove redundent withMaterializedStatementView/withNewMaterializedView methods in StarRocksAssert (#16985)

Signed-off-by: Kevin Li <ming.moriarty@gmail.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

Commit:
[Doc] Add Upgrade Info (#16988)

Signed-off-by: EsoragotoSpirit <wanglichen@starrocks.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

nshangyiming committed:
[Enhancement] Refine the error msg of drop func and other priv checking (#16980)

Signed-off-by: Dejun Xia <xiadejun@starrocks.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

silverbullet233 committed:
[Enhancement] add MemUsageBytes when showing current queries (#16923)



Signed-off-by: silverbullet233 <3675229+silverbullet233@users.noreply.github.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

nshangyiming committed:
[BugFix] Restore logging for cluster load statistics (#17000)

The logging for cluster load statistics is mistakenly removed by previous pr.

Signed-off-by: HangyuanLiu <460660596@qq.com>

murphyatwork committed:
[Enhancement] remove noavx flags for simdjson (#16896)

Signed-off-by: HangyuanLiu <460660596@qq.com>

xiaoyong-z committed:
[Enhancement] Too many register compaction task failed error msg (#16766)

If we set max_compaction_concurrency=0, there will be too many useless
register compaction task failed log.
In this pr, we only print this error msg at first time.

Signed-off-by: xyz <a997647204@gmail.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

xiaoyong-z committed:
[Refactor] optimize jindosdk decompress procedure (#17028)

Currently, we will decompress jindosdk-4.6.2.tar.gz when building broker,
this is very time consuming.

In this pr, we optimize this procedure, jindosdk-4.6.2.tar.gz will
only be decompressed in the first time it downloads.

Signed-off-by: xyz <a997647204@gmail.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

satanson committed:
[BugFix] Offset+limit overflows and yields a negative result (#16922)

Signed-off-by: satanson <ranpanf@gmail.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

Youngwb committed:
[BugFix] Fix show materialized view do not contains the mv which base table is external table (#16971)

Signed-off-by: Youngwb <yangwenbo_mailbox@163.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

abc982627271 committed:
[Feature]Multi cluster warehouse  (#16728)

Multi cluster warehouse

Signed-off-by: HangyuanLiu <460660596@qq.com>

decster committed:
[Enhancement] Add info about max_countinous_version in some log (#17008)

Signed-off-by: Binglin Chang <decstery@gmail.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

sduzh committed:
[Feature] Support schema changes to the lake table of primary key (#16850)


Signed-off-by: sduzh <zhuming9011@qq.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

Commit:
[Doc] Rectify Unit desc (#17039)

Signed-off-by: EsoragotoSpirit <wanglichen@starrocks.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

Commit:
[Doc] Rectify MV Drop column process (#17046)

Signed-off-by: EsoragotoSpirit <wanglichen@starrocks.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

fzhedu committed:
[Refactor] refactor auto streaming agg to avoid worst cases (#15939)

refactor auto streaming agg to avoid worst cases

Signed-off-by: HangyuanLiu <460660596@qq.com>

starrocks-xupeng committed:
[Feature] support lake colocate table DDL (#16956)

Signed-off-by: starrocks-xupeng <xupeng@starrocks.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

ABingHuang committed:
fix unstable union mv rewrite ut

Signed-off-by: ABingHuang <codekhuang@163.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

ABingHuang committed:
fix compile

Signed-off-by: ABingHuang <codekhuang@163.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

nshangyiming committed:
[BugFix] Fix load rbac image failure (#17081)

Fix load rbac image failure 

Signed-off-by: HangyuanLiu <460660596@qq.com>

Seaven committed:
[Enhancment] disable unstable ut (#17074)

Signed-off-by: Seaven <seaven_7@qq.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

trueeyu committed:
remove unused config of BE (#17065)

Signed-off-by: trueeyu <lxhhust350@qq.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

liuyehcf committed:
[BugFix] Fix assigning wrong type for empty array/map/struct (#16658)

Signed-off-by: HangyuanLiu <460660596@qq.com>

HangyuanLiu committed:
[BugFix] The db_admin role miss some privilege type in new RBAC (#16932)

* The db_admin role miss some privilege type in new RBAC

Signed-off-by: HangyuanLiu <460660596@qq.com>

ABingHuang committed:
fix ut

Signed-off-by: HangyuanLiu <460660596@qq.com>

ABingHuang committed:
modify name

Signed-off-by: ABingHuang <codekhuang@163.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

evelynzhaojie committed:
[Doc] add behavior change to 2.5 rn (#17084)

* [Doc] add behavior change to 2.5 rn

Signed-off-by: evelynzhaojie <everlyn.zhaojie@gmail.com>

* update of system variables

Signed-off-by: evelynzhaojie <everlyn.zhaojie@gmail.com>

---------

Signed-off-by: evelynzhaojie <everlyn.zhaojie@gmail.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

xiaoyong-z committed:
[Refactor] change the permission mode of tool script file (#17053)

Signed-off-by: HangyuanLiu <460660596@qq.com>

sduzh committed:
[Enhancement] Create lake table by default if use_staros is true (#16962)

Signed-off-by: sduzh <zhuming9011@qq.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

silverbullet233 committed:
[Enhancement] add PruneGroupByKeysRule (#16716)



Signed-off-by: silverbullet233 <3675229+silverbullet233@users.noreply.github.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

Youngwb committed:
[BugFix] Fix query rewrite based on MV failed after refresh MV (#17067)

Signed-off-by: Youngwb <yangwenbo_mailbox@163.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

Astralidea committed:
[Feature] Support list partition (#17037)

* [Feature] Support list partition

Signed-off-by: Astralidea <astralidea@163.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

amber-create committed:
[Doc] fix Keywords links (#17088)

Signed-off-by: amber-create <yangyanping@starrocks.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

nshangyiming committed:
[BugFix] Handle exception on obj not found when checking priv (#17082)

Signed-off-by: Dejun Xia <xiadejun@starrocks.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

Seaven committed:
[Enhancement] Remove enforce task re-compute statistics (#15358)

Remove enforce task re-compute statistics

Signed-off-by: HangyuanLiu <460660596@qq.com>

Seaven committed:
[BugFix] Complex shuffle join with aggregate bug (#16834)

Signed-off-by: Seaven <seaven_7@qq.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

trueeyu committed:
[Tool] Add benchmark case for shuffle chunk on large cluster (#16990)

* add benchmark case for shuffle chunk on large cluster

Signed-off-by: trueeyu <lxhhust350@qq.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

stdpain committed:
[Enhancement] Make Aggregator release memory in close stage (#16913)

now aggregator will release map in Aggregator::~Aggregator(). which
won't release memory until QueryContext call destructor

Signed-off-by: stdpain <drfeng08@gmail.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

stephen-shelby committed:
[Feature] select into outfile support parquet format (#16396)

Signed-off-by: HangyuanLiu <460660596@qq.com>

Astralidea committed:
[Feature] Import the automatic partition creation metadata part (#16524)

* Import the automatic partition creation metadata part

Signed-off-by: Astralidea <astralidea@163.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

amber-create committed:
[Doc] update sort key doc (#17112)

* [Doc] update sort key doc
Signed-off-by: amber-create <yangyanping@starrocks.com>

* Update Sort_key.md

Signed-off-by: HangyuanLiu <460660596@qq.com>

abc982627271 committed:
update code (#17109)

Signed-off-by: abc982627271 <liuxuefen@starrocks.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

HangyuanLiu committed:
Decoupling ObjectType and PrivilegeType in RBAC privilege (#17123)

Decoupling ObjectType and PrivilegeType in RBAC privilege

Signed-off-by: HangyuanLiu <460660596@qq.com>

satanson committed:
[BugFix] Apply ReduceCastRule to child exprs of SetOperationRelation (#16948)

RelationTransformer forgets to apply ReduceCastRule to child exprs of SetOperationRelation, so a cast expr(something like cast(varchar(-1) as varchar(2)) is propagated to BE, and VectorizedCastToString::_evaluate_string is invoked to process this cast expr and the input strings are truncated unexpectedly.

Signed-off-by: HangyuanLiu <460660596@qq.com>

packy92 committed:
[Enhancement] support trace verbose optimizer and add limit to semi join reorder (#17070)

Signed-off-by: HangyuanLiu <460660596@qq.com>

dirtysalt committed:
[Feature] Support AMS(Array/Map/Struct) in Hudi MOR on C++ side (#16925)

Signed-off-by: HangyuanLiu <460660596@qq.com>

Seaven committed:
[UT] Fix error ut (#17140)

Fix error ut 

Signed-off-by: HangyuanLiu <460660596@qq.com>

ABingHuang committed:
ignore mv rewrite ut temporarily to fix bug (#17148)

Signed-off-by: ABingHuang <codekhuang@163.com>
Signed-off-by: HangyuanLiu <460660596@qq.com>

wanpengfei-git commented:
#### [FE PR Coverage Check]

:disappointed: **fail** : 128 / 425 (30.12%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/privilege/UserPEntryObject.java|0|1|00.00%|[34]|
|:large_blue_circle:|com/starrocks/sql/ast/GrantPrivilegeStmt.java|0|2|00.00%|[41, 43]|
|:large_blue_circle:|com/starrocks/privilege/ResourcePEntryObject.java|0|1|00.00%|[29]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeCollection.java|0|3|00.00%|[62, 66, 318]|
|:large_blue_circle:|com/starrocks/server/CatalogMgr.java|0|1|00.00%|[277]|
|:large_blue_circle:|com/starrocks/privilege/DbPEntryObject.java|0|1|00.00%|[32]|
|:large_blue_circle:|com/starrocks/privilege/FunctionPEntryObject.java|0|2|00.00%|[38, 42]|
|:large_blue_circle:|com/starrocks/privilege/GlobalFunctionPEntryObject.java|0|1|00.00%|[33]|
|:large_blue_circle:|com/starrocks/privilege/ResourceGroupPEntryObject.java|0|1|00.00%|[32]|
|:large_blue_circle:|com/starrocks/privilege/TablePEntryObject.java|0|2|00.00%|[37, 41]|
|:large_blue_circle:|com/starrocks/privilege/CatalogPEntryObject.java|0|1|00.00%|[42]|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|10|145|06.90%|[535, 536, 537, 538, 539, 540, 541, 1724, 1725, 1727, 1728, 1730, 1731, 1732, 1735, 1736, 1738, 1739, 1740, 1741, 1742, 1744, 1745, 1746, 1747, 1749, 1750, 1752, 1753, 1754, 1755, 1756, 1758, 1759, 1760, 1761, 1762, 1764, 1765, 1766, 1767, 1768, 1769, 1770, 1772, 1773, 1775, 1776, 1779, 1780, 1781, 1782, 1783, 1784, 1786, 1787, 1788, 1789, 1790, 1791, 1793, 1794, 1795, 1797, 1798, 1799, 1800, 1802, 1803, 1804, 1805, 1807, 1810, 1811, 1813, 1814, 1815, 1816, 1818, 1819, 1820, 1821, 1822, 1824, 1825, 1826, 1827, 1828, 1829, 1830, 1832, 1833, 1835, 1837, 1838, 1839, 1841, 1842, 1843, 1844, 1846, 1847, 1850, 1851, 1852, 1853, 1854, 1855, 1857, 1858, 1859, 1861, 1863, 1864, 1865, 1866, 1867, 1868, 1875, 1877, 1878, 1879, 1881, 1882, 1883, 1884, 1885, 1886, 1887, 1889, 1891, 1892, 1893, 1895, 1896]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeCheckerV2.java|3|19|15.79%|[1052, 1053, 1055, 1056, 1058, 1059, 1062, 1063, 1064, 1066, 1067, 1068, 1091, 1092, 1093, 1118]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AstToStringBuilder.java|34|137|24.82%|[293, 314, 315, 316, 318, 321, 322, 323, 324, 326, 327, 328, 330, 334, 335, 336, 338, 339, 340, 341, 343, 345, 346, 347, 350, 353, 354, 355, 357, 358, 359, 361, 364, 365, 368, 369, 370, 372, 373, 374, 376, 379, 380, 381, 383, 384, 386, 389, 390, 391, 393, 395, 396, 397, 398, 399, 400, 401, 403, 404, 406, 409, 411, 412, 413, 415, 416, 418, 419, 420, 421, 423, 424, 425, 428, 430, 433, 434, 435, 436, 438, 439, 440, 443, 446, 447, 448, 449, 450, 452, 453, 455, 459, 460, 462, 464, 465, 467, 470, 476, 498, 503, 528]|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|3|33.33%|[596, 600]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowRolesStmt.java|3|6|50.00%|[51, 52, 54]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowGrantsStmt.java|11|21|52.38%|[63, 64, 65, 73, 74, 75, 76, 97, 98, 100]|
|:large_blue_circle:|com/starrocks/sql/ast/BaseGrantRevokePrivilegeStmt.java|2|3|66.67%|[151]|
|:large_blue_circle:|com/starrocks/privilege/PrivilegeManager.java|7|10|70.00%|[1307, 1329, 1358]|
|:large_blue_circle:|com/starrocks/sql/ast/ShowUserStmt.java|4|5|80.00%|[45]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|25|30|83.33%|[3992, 3993, 3995, 4013, 4014]|
|:large_blue_circle:|com/starrocks/privilege/DefaultAuthorizationProvider.java|5|6|83.33%|[74]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeStmtAnalyzerV2.java|14|15|93.33%|[359]|
|:large_blue_circle:|com/starrocks/sql/ast/GrantRoleStmt.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/RevokeRoleStmt.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/mysql/privilege/Auth.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeStmtAnalyzer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/BaseGrantRevokeRoleStmt.java|2|2|100.00%|[]|

HangyuanLiu committed:
Merge remote-tracking branch 'upstream/main' into show_grants

# Conflicts:
#	fe/fe-core/src/main/java/com/starrocks/privilege/DefaultAuthorizationProvider.java
#	fe/fe-core/src/main/java/com/starrocks/privilege/PrivilegeManager.java
#	fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java
#	fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeCheckerV2.java

Signed-off-by: HangyuanLiu <460660596@qq.com>
